internal/provider/resource_ko_resolve.go

```diff
@@ -0,0 +1,263 @@
+				Optional:    true,
+				Type:        schema.TypeBool,
+				ForceNew:    true,
+			},
+			PushKey: {
+				Description: "Push images to KO_DOCKER_REPO",
+				Default:     true,
+				Optional:    true,
+				Type:        schema.TypeBool,
+				ForceNew:    true,
+			},
+			SelectorKey: {
+				Description: "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)",
+				Optional:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+			PlatformsKey: {
+				Description: "Which platform to use when pulling a multi-platform base. Format: all | <os>[/<arch>[/<variant>]][,platform]*",
+				Optional:    true,
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				ForceNew:    true,
+			},
+			SBOMKey: {
+				Description: "The SBOM media type to use (none will disable SBOM synthesis and upload, also supports: spdx, cyclonedx, go.version-m).",
+				Default:     "spdx",
+				Optional:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+			BaseImageKey: {
+				Description: "",
+				Default:     defaultBaseImage,
+				Optional:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+			TagsKey: {
+				Description: "Which tags to use for the produced image instead of the default 'latest' tag ",
+				Optional:    true,
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				ForceNew:    true,
+			},
+
+			// Computed fields
+			ManifestsKey: {
+				Description: "A list of resolved manifests in a 'yamldecode'able format. Note that whitespaces and nil docs will be stripped from these results.",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Computed:    true,
+			},
+		},
+	}
+}
+
+type Resolver interface {
+	Resolve(ctx context.Context) (*Resolved, error)
+}
+
+type Resolved struct {
+	Id        string
+	Manifests []string
+}
+
+type resolver struct {
+	bo *options.BuildOptions
+	po *options.PublishOptions
+	fo *options.FilenameOptions
+	so *options.SelectorOptions
+}
+
+func NewResolver(d *schema.ResourceData, meta interface{}) (*resolver, error) {
+	opts, err := NewProviderOpts(meta)
+	if err != nil {
+		return nil, err
+	}
+
+	r := &resolver{
+		bo: opts.bo,
+		po: opts.po,
+		fo: &options.FilenameOptions{},
+		so: &options.SelectorOptions{},
+	}
+
+	if p, ok := d.Get(BaseImageKey).(string); ok {
+		r.bo.BaseImage = p
+	}
+
+	if p, ok := d.Get(TagsKey).([]interface{}); ok {
+		if len(p) == 0 {
+			r.po.Tags = []string{"latest"}
+		} else {
+			r.po.Tags = StringSlice(p)
+		}
+	}
+
+	if p, ok := d.Get(PushKey).(bool); ok {
+		r.po.Push = p
+	}
+
+	if p, ok := d.Get(FilenamesKey).([]interface{}); ok {
+		r.fo.Filenames = StringSlice(p)
+	}
+
+	if p, ok := d.Get(RecursiveKey).(bool); ok {
+		r.fo.Recursive = p
+	}
+
+	if p, ok := d.Get(SelectorKey).(string); ok {
+		r.so.Selector = p
+	}
```

Comment:
I just YOLO'ed all these in resource_ko_image.go and let it panic if the wrong type came through. I thought we were trusting terraform internals to typecheck these for us?